### ISSUE TITLE
How will we make make math content accessible?

### ISSUE THREAD

**Issue by terracoda** [phetsims/a11y-research#106]
I am starting a broader issue on Accessible Math where we can gather questions, working solutions, tools and research that we come across as we start to dive into sims that contain math content.

@jonathanolson is doing some testing with MathML. There are questions about the scientific notation in Coulomb's Law and Gravity Force Lab: Basics see https://github.com/phetsims/a11y-research/issues/105.

I’m not sure when we are going to dive into math accessibility and do a literature review or find existing tools that might work across all browsers. I think everyone is aware that Chrome does not support MathML, so it may not be our long term solution.

The top 2 math experts in accessible math that I am aware of are TV Raman and Volker Sorge. Both have done extensive work in making Math accessible. TVRaman on EmacsSpeaks and Volker Sorge on MathJax.

**Gathering a few Resources, Tools, and Reading**
- [MathJax](https://www.mathjax.org/)
- [EmacsSpeaks](https://github.com/tvraman/emacspeak//blob/master/lisp/emacspeak-sage.el#L108)
- [Thinking of Mathematics](http://emacspeak.sourceforge.net/raman/publications/thinking-of-math/thinking-of-math.html)
- [New Accessibility Features in MathJax](http://scholarworks.csun.edu/bitstream/handle/10211.3/180124/ID-31-Cervone-JTPD-2016.pdf?sequence=1)

Everyone, please add to the list if you find something of interest!!

@jonathanolson, could share your progress (or questions/issues) with MathML in this issue? I am assigning to you for now.

**Comment by jonathanolson** [comment]
Current MathML results are noted in https://github.com/phetsims/QA/issues/121.

**Comment by jonathanolson** [comment]
Additionally, we'll probably want to review the testing done in https://github.com/phetsims/QA/issues/121, since it seems NVDA may be an issue for using MathML. I'll definitely be interested in the proper solution for this.

**Comment by jessegreenberg** [comment]
@jonathanolson I have heard you mention that it is difficult to integrate MathJax into scenery, can you please summarize the difficulties?

**Comment by jonathanolson** [comment]
> @jonathanolson I have heard you mention that it is difficult to integrate MathJax into scenery, can you please summarize the difficulties?

It has a very hard-coded "startup and load external resources" style. It depends on external CSS/etc. files, including dynamically loading images, fonts, etc. depending on what is needed and what the given platform is. See https://github.com/phetsims/scenery/issues/457

It would take a lot of work (40-80 hours?) to fully patch MathJax so that it could be included in a sim, and more time to figure out how each sim would specify the subset of "dynamically loaded files" that we'd have to data-URL-include in the built file. It would potentially add a large amount of file size to sims that use it (could double/triple), and it would include a lot of maintenance costs.

If we'd just want to "use it" for accessibility, I think the better approach would be to look into what they do for a11y and how the do it. If it uses a library just for that, then maybe there's something we could use directly.

**Comment by jessegreenberg** [comment]
Thanks, that is very helpful.

**Comment by terracoda** [comment]
@jonathanolson, yes thanks, that is very helpful information!

**Comment by jessegreenberg** [comment]
@terracoda I saw you added the meeting label, would you like to discuss this at an upcoming a11y meeting?

**Comment by terracoda** [comment]
@jessegreenberg, the meeting label was added 3 weeks ago. We had the FYI meeting. I think we will explore this further as we move forward with math sims. I'll label this onhold for now?

Please change label if not appropriate.

**Comment by terracoda** [comment]
Oops only meant to put onhold, not close.

**Comment by jessegreenberg** [comment]
Sounds good, thanks!

**Comment by emily-phet** [comment]
Adding some resources from Paul at Mind Research Institute:

Some very simple test cases for MathML and its alternatives, with and without MathJax.  Mostly just to test speech and braille output:
http://superkeen.com/web/mathmltest/
 
Test cases for MathML embedded within other HTML elements:
http://superkeen.com/web/mathmltest/mathmlfocus.html
 
Related Orca MathML issue:
https://gitlab.gnome.org/GNOME/orca/issues/26
 
Pixis JS’s accessibility feature with debugging on, for easy inspection:
http://superkeen.com/web/pixi/index-debug.html
 
Same thing but with the production settings:
http://superkeen.com/web/pixi/
 
Related Pixi improvements:
https://github.com/pixijs/pixi.js/pull/4435

**Comment by terracoda** [comment]
Browser support for MathML is better now,  [https://caniuse.com/mathml](https://caniuse.com/mathml)

And here are the resources listed at caniuse.com
- [MathML Specification](https://www.w3.org/TR/MathML/) [w3.org]
- [MDN Web Docs - MathML](https://developer.mozilla.org/en-US/docs/Web/MathML) [developer.mozilla.org]
- [Wikipedia](https://en.wikipedia.org/wiki/MathML) [en.wikipedia.org]
- [MathML torture test](https://fred-wang.github.io/MathFonts/mozilla_mathml_test/) [fred-wang.github.io]
- [MathML in Chromium Project Roadmap](https://mathml.igalia.com/project/) [mathml.igalia.com]
- [Cross-browser support script](https://www.mathjax.org/) [mathjax.org]
- [Chrome platform status: Enabled by default](https://www.chromestatus.com/feature/5240822173794304) [chromestatus.com]ources listed caniuse.com for mathml:
